### PR TITLE
Move NextLink inside Control

### PR DESCRIPTION
**Pull request by davidpmccormick**
Fixing invalid markup (an `<a>` shouldn't contain a `<button>` as a descendant).

### CODE REVIEW

File: catalogue/webapp/pages/item.js [hunk 1]
```diff
@@ -243,22 +243,29 @@ const PaginatorButtons = ({
...
-              [spacing({ s: 1 }, { margin: ['right'] })]: true,
-            })}
-          >
-            <Control type="light" icon="arrow" extraClasses="icon--180" />
-          </a>
-        </NextLink>
+        <Control
+          scroll={false}
+          replace={true}
+          nextLink={prevLink}
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/item.js]: I think we've just called it link elsewhere.

Comment (davidpmccormick) [reply catalogue/webapp/pages/item.js]: the prop, or the component? Or both?

File: common/views/components/Buttons/Control/Control.js [hunk 2]
```diff
@@ -70,10 +76,25 @@ const Control = forwardRef(
...
     return (
-      // $FlowFixMe (conditional JSX tag)
-      <HtmlTag ref={ref} {...attrs}>
-        <InnerControl text={text} icon={icon} />
-      </HtmlTag>
+      <>
+        {nextLink ? (
+          <NextLink
+            {...nextLink}
+            scroll={scroll}
```
Comment (jamesgorrie) [inline common/views/components/Buttons/Control/Control.js]: Not sure what is easier to read, but was thinking you can use this syntax:
```jsx 
<NextLink
            {...nextLink}
            {... {scroll, replace, prefetch }}
          />
```

File: common/views/components/Buttons/Control/Control.js [hunk 3]
```diff
@@ -70,10 +76,25 @@ const Control = forwardRef(
...
+            replace={replace}
+            prefetch={prefetch}
+          >
+            <a ref={ref} {...attrs}>
+              <InnerControl text={text} icon={icon} />
+            </a>
+          </NextLink>
+        ) : (
+          // $FlowFixMe (conditional element)
```
Comment (jamesgorrie) [inline common/views/components/Buttons/Control/Control.js]: What's this?

File: catalogue/webapp/pages/item.js (hunk 1 above)
Comment (jamesgorrie) [reply catalogue/webapp/pages/item.js]: The `prop`.

File: common/views/components/Buttons/Control/Control.js (hunk 3 above)
Comment (davidpmccormick) [reply common/views/components/Buttons/Control/Control.js]: I'm not sure (it was pre-existing rage-`$flowFixMe`). It didn't make sense to me, but it's possible I'd overlooked something.

File: common/views/components/Buttons/Control/Control.js (hunk 2 above)
Comment (davidpmccormick) [reply common/views/components/Buttons/Control/Control.js]: I think I'ma leave it as is unless you object – I think it looks nice, but I find it considerably harder to wrap my head around!

Comment (jamesgorrie) [reply common/views/components/Buttons/Control/Control.js]: No objection, just a thought.

**Pull request merged**